### PR TITLE
Add XQuAD dataset in Retrieval task

### PR DESCRIPTION
**Pull request by jaygala24**
## Checklist for adding MMTEB dataset

Reason for dataset addition:

- [x] I have tested that the dataset runs with the `mteb` package.
- [x] I have run the following models on the task (adding the results to the pr). These can be run using the `mteb run -m {model_name} -t {task_name}` command.
  - [x] `sentence-transformers/paraphrase-multilingual-MiniLM-L12-v2`
  - [x] `intfloat/multilingual-e5-small`
- [x] I have checked that the performance is neither trivial (both models gain close to perfect scores) nor random (both models gain close to random scores).
- [x] If the dataset is too big (e.g. >2048 examples), considering using `self.stratified_subsampling() under dataset_transform()`
- [x] I have filled out the metadata object in the dataset file (find documentation on it [here](https://github.com/embeddings-benchmark/mteb/blob/main/docs/adding_a_dataset.md#2-creating-the-metadata-object)).
- [x] Run tests locally to make sure nothing is broken using `make test`. 
- [x] Run the formatter to format the code using `make lint`. 
- [x] I have added points for my submission to the [points folder](https://github.com/embeddings-benchmark/mteb/blob/main/docs/mmteb/points.md) using the PR number as the filename (e.g. `438.jsonl`).

### CODE REVIEW

File: mteb/tasks/Retrieval/multilingual/XQuADRetrieval.py [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+                question = row["question"]
+                context = row["context"]
+                query_id = question_ids[question]
+                queries[lang][split][query_id] = question
+
+                doc_id = context_ids[context]
+                corpus[lang][split][doc_id] = {"text": context}
+                if query_id not in relevant_docs[lang][split]:
+                    relevant_docs[lang][split][query_id] = {}
+                relevant_docs[lang][split][query_id][doc_id] = 1
```
Comment (imenelydiaker) [inline mteb/tasks/Retrieval/multilingual/XQuADRetrieval.py:106]: I think you can simplify this by using the `id` provided in the dataset, and build an id for queries by enumerating them (this dataset exists for long enough I'm not sure it contains duplicates). See example [here](https://github.com/embeddings-benchmark/mteb/blob/main/mteb/tasks/Retrieval/fra/FQuADRetrieval.py) and [here](https://github.com/embeddings-benchmark/mteb/blob/main/mteb/tasks/Retrieval/fra/SyntecRetrieval.py)

File: mteb/tasks/Retrieval/multilingual/XQuADRetrieval.py [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+        reference="https://huggingface.co/datasets/xquad",
+        type="Retrieval",
+        category="s2p",
+        eval_splits=["validation"],
+        eval_langs=_LANGUAGES,
+        main_score="ndcg_at_10",
+        date=("2019-05-21", "2019-11-21"),
+        form=["written"],
+        domains=["Web"],
+        task_subtypes=[],
```
Comment (imenelydiaker) [inline mteb/tasks/Retrieval/multilingual/XQuADRetrieval.py:45]: Any guess about this field? Mayeb you can check the paper?

Comment (jaygala24) [reply mteb/tasks/Retrieval/multilingual/XQuADRetrieval.py]: This dataset was proposed for question task so I have updated the metadata accordingly.

File: mteb/tasks/Retrieval/multilingual/XQuADRetrieval.py (hunk 1 above)
Comment (jaygala24) [reply mteb/tasks/Retrieval/multilingual/XQuADRetrieval.py]: I have made the necessary revisions as suggested in your comments above.

**Review by imenelydiaker — APPROVED** [review]
LGTM! Thank you for this addition, please add points and we'll be able to merge 🙂

**Comment by jaygala24** [comment]
> LGTM! Thank you for this addition, please add points and update your branch, after that we'll be able to merge 🙂

@imenelydiaker 

This dataset adds 5 new languages (ara, ell, ron, rus, tha) to the Retrieval task. The total points would be 2 + 4 x 5 = 22 pts. I have added the points in the latest commit.

**Comment by imenelydiaker** [comment]
Okay my bad I thought we counted 2 points per language x dataset.
So it's 2 + 4 * 2 = 10 since other languages are already covered. Thank you again for this contribution!

**Comment by jaygala24** [comment]
> Okay my bad I thought we counted 2 points per language x dataset. So it's 2 + 4 * 2 = 10 since other languages are already covered. Thank you again for this contribution!

@imenelydiaker I have updated the points in the latest commit.

**Pull request merged**